Source/Applications/SystemCenter/wwwroot/Scripts/TSX/SystemCenter/ExternalDB/ExternalDBForm.tsx

```diff
@@ -22,45 +22,73 @@
 //******************************************************************************************************
 
 import * as React from 'react';
-import { SystemCenter } from '@gpa-gemstone/application-typings';
+import { Application, SystemCenter } from '@gpa-gemstone/application-typings';
 import { Input, TextArea, CheckBox } from '@gpa-gemstone/react-forms';
 import { IsCron } from '@gpa-gemstone/helper-functions';
+import { LoadingScreen, Modal } from '@gpa-gemstone/react-interactive';
 
-export default function ExternalDBForm(props: { Record: SystemCenter.Types.ExternalDatabases, Setter: (record: SystemCenter.Types.ExternalDatabases) => void, setErrors?: (e: string[]) => void }) {
+export default function ExternalDBForm(props: {
+    Record: SystemCenter.Types.ExternalDatabases,
+    Setter: (record: SystemCenter.Types.ExternalDatabases) => void,
+    setErrors?: (e: string[]) => void,
+    ShowTestButton?: boolean
+}) {
 
-    const [showDataProvider, setShowDataProvider] = React.useState<boolean>(false);
+    const [requestStatus, setRequestStatus] = React.useState<Application.Types.Status>('unintiated');
 
     function Valid(field: keyof (SystemCenter.Types.ExternalDatabases)): boolean {
         if (field == 'Name')
             return props.Record.Name != null && props.Record.Name.length > 0 && props.Record.Name.length <= 200;
         else if (field == 'Schedule')
-            return props.Record.Schedule != null && IsCron(props.Record.Schedule);
+            return (props.Record.Schedule == null || props.Record.Schedule.length == 0) || (props.Record.Schedule != null && IsCron(props.Record.Schedule));
         else if (field == 'ConnectionString' || field == 'DataProviderString')
             return true;
         return false;
     }
 
-    function DataProvider() {
-        if (showDataProvider) {
-            return <TextArea<SystemCenter.Types.ExternalDatabases> Rows={3} Record={props.Record} Field={'DataProviderString'} Valid={Valid} Setter={props.Setter} />
-        }
-
-        return null;
-    }
-
-    React.useEffect(() => {
-        setShowDataProvider(props.Record.Encrypt);
-    }, [props.Record.Encrypt]);
+    const TestExternal = React.useCallback(() => {
+        setRequestStatus('loading');
+        let handle = $.ajax({
+            type: "POST",
+            url: `api/SystemCenter/ExternalDatabases/TestConnection`,
+            contentType: "application/json; charset=utf-8",
+            data: JSON.stringify(props.Record),
+            dataType: 'json',
+            cache: false,
+            async: true
+        });
+        handle.done(() => {
+            setRequestStatus('idle');
+        });
+        handle.fail(() => {
+            setRequestStatus('error');
+        });
+        return () => {
+            if (handle != null && handle.abort != null) handle.abort();
+        };
+    }, [props.Record, setRequestStatus]);
 
+    const ClosePopup = React.useCallback(() => {
```

Comment:
that is unneccesarry, state Setters are not changeing Ref so you can use them directly